Add FontCollection tests and fix compat bugs

hughbe opened the pull request.

dnfclas commented:
@hughbe,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

hughbe commented on src/font.c:
```diff
@@ -159,6 +161,13 @@ GdipPrivateAddFontFile (GpFontCollection *font_collection,  GDIPCONST WCHAR *fil
 	if (!file)
 		return OutOfMemory;
 
+	fileHandle = fopen ((char *)file, "r");
```
`GdipPrivateAddFontFile` returns `FileNotFound` if the file doesn't exist with GDI+. Previously libgdiplus returned `Ok`

hughbe commented on src/font.c:
```diff
@@ -290,13 +299,13 @@ GdipGetFontCollectionFamilyList (GpFontCollection *font_collection, INT num_soug
 		gdip_createPrivateFontSet (font_collection);
 
-	for (i = 0; i < font_collection->fontset->nfont; i++) {
+	for (i = 0; i < num_sought && i < font_collection->fontset->nfont; i++) {
```
Here (and below), libgdiplus ignored the `num_sought` parameter. This now matches GDI+ behaviour, obviously.

hughbe commented on src/font.c:
```diff
@@ -1152,14 +1160,12 @@ GdipPrivateAddMemoryFont(GpFontCollection *fontCollection, GDIPCONST void *memor
 	close(f);
 
-	/* FIXME - this doesn't seems to catch "bad" (e.g. invalid) font files */
-	result = FcConfigAppFontAddFile (fontCollection->config, fontfile);
```
It appears that GDI+ just returns Ok for invalid files (added a test for that). Previously, libgdiplus would seemingly non-deterministically return `FileNotFound`.

In both libraries the count stays at the previous number if the font file is invalid.

akoeplinger commented on src/font.c:
```diff
@@ -159,6 +161,13 @@ GdipPrivateAddFontFile (GpFontCollection *font_collection,  GDIPCONST WCHAR *fil
 		return OutOfMemory;
 
+	fileHandle = fopen ((char *)file, "r");
+	if(!fileHandle) {
```
nit: space before `(`

akoeplinger commented on src/font.c:
```diff
@@ -159,6 +161,13 @@ GdipPrivateAddFontFile (GpFontCollection *font_collection,  GDIPCONST WCHAR *fil
+		return FileNotFound;
+	}
+
+	fclose(fileHandle);
```
nit: space before `(`

hughbe commented:
I've addressed the PR feedback.

> Where did you get the test.ttf and test.otf fonts from, are they open source?

I got them from the [corefx-testdata](https://github.com/dotnet/corefx-testdata) repository. I recall that I created the font files via a tool on my PC.

hughbe commented:
Yes I am!
